desc/magnetic_fields.py

```diff
@@ -59,6 +64,81 @@ def biot_savart(eval_pts, coil_pts, current):
     return B
 
 
+def read_BNORM_file(fname, surface, eval_grid=None, scale_by_curpol=True):
+    """Read BNORM-style .txt file containing Bnormal Fourier coefficients.
+
+    Parameters
+    ----------
+    fname : str
+        name of BNORM file to read and calculate Bnormal from.
+    surface : Surface or Equilibrium
+        Surface to calculate the magnetic field's Bnormal on.
+        If an Equilibrium is supplied, will use its boundary surface.
+    eval_grid : Grid, optional
+        Grid of points on the plasma surface to calculate the Bnormal at,
+        if None defaults to a LinearGrid with twice
+        the surface grid's poloidal and toroidal resolutions
+    scale_by_curpol : bool, optional
+        Whether or not to un-scale the Bnormal coefficients by curpol
+        before calculating Bnormal, by default True
+        (set to False if it is known that the BNORM file was saved without scaling
+        by curpol)
+        requires an Equilibrium to be passed in
+
+
+    Returns
+    -------
+    Bnormal: ndarray,
+        Bnormal distribution from the BNORM Fourier coefficients,
+        evaluated on the given eval_grid
+    """
+    if isinstance(surface, EquilibriaFamily):
+        surface = surface[-1]
+    if isinstance(surface, Equilibrium):
+        eq = surface
+        surface = eq.surface
+    else:
+        eq = None
+
+    if scale_by_curpol and eq is None:
+        raise RuntimeError(
+            "an Equilibrium must be supplied when scale_by_curpol is True!"
+        )
+
+    curpol = (
+        (2 * jnp.pi / eq.NFP * eq.compute("G", grid=LinearGrid(rho=jnp.array(1)))["G"])
+        if scale_by_curpol
+        else 1
+    )
+
+    data = np.genfromtxt(fname)
+
+    xm = data[:, 0]
+    xn = -data[:, 1]  # negate since BNORM uses sin(mu+nv) convention
+    Bnorm_mn = data[:, 2] / curpol  # these will only be sin terms
+
+    # convert to DESC Fourier representation i.e. like cos(mt)*cos(nz)
+    m, n, Bnorm_mn = ptolemy_identity_fwd(xm, xn, Bnorm_mn, np.zeros_like(Bnorm_mn))
+    basis = DoubleFourierSeries(
+        int(np.max(m)), int(np.max(n)), sym=False, NFP=surface.NFP
```

Comment:
Shouldn't this be symmetric?

Reply:
yea I think I had it as false before I fixed another bug we had in the VMEC mode stuff, changed it to sym = 'sin' now